src/elexmodel/client.py

```diff
@@ -91,9 +91,11 @@ def _check_input_parameters(
             raise ValueError("beta is not valid. Has to be either an integer or a float.")
         if not isinstance(robust, bool):
             raise ValueError("robust is not valid. Has to be a boolean.")
-        if not isinstance(lambda_, (float, int)):
-            raise ValueError("lambda is not valid. It has to be numeric.")
-        if lambda_ < 0:
+        if not isinstance(lambda_, list):
+            raise ValueError("lambda is not valid. It has to be a list.")
+        if not all(isinstance(e, (int, float)) for e in lambda_) and len(lambda_) > 0:
```

Comment:
I think turning this around makes a bit more sense:
```
if len(lambda_) > 0 and not all(isinstance(e, (int, float)) for e in lambda_):
```
Since if `len(lambda_) < 0` this will short circuit and the second part will never get checked.